packages/state-manager/src/sagas/identity/identity.selectors.ts

```diff
@@ -37,6 +38,22 @@ export const csr = createSelector(communitiesSelectors.currentCommunityId, selec
   return identities[id]?.userCsr
 })
 
+export const usernameTaken = createSelector(currentIdentity, certificatesMapping, (identity, certs) => {
+  const userCertificate = identity?.userCertificate
+  if (userCertificate) return false
+
+  const username = identity?.nickname
+  if (!username) return false
```

Comment:
Should this be `if (username) return false`?

Reply:
no, this is for TS mainly, and what you're proposing is that if user picked any username that means that this username is not taken which is obviously not true.